python/ray/tune/suggest/optuna.py

```diff
@@ -189,21 +192,17 @@ def suggest(self, trial_id: str) -> Optional[Dict]:
                     mode=self._mode))
 
         if trial_id not in self._ot_trials:
-            ot_trial_id = self._storage.create_new_trial(
-                self._ot_study._study_id)
-            self._ot_trials[trial_id] = ot.trial.Trial(self._ot_study,
-                                                       ot_trial_id)
+            self._ot_trials[trial_id] = self._ot_study.ask()
+
         ot_trial = self._ot_trials[trial_id]
 
-        if self._points_to_evaluate:
-            params = self._points_to_evaluate.pop(0)
-        else:
-            # getattr will fetch the trial.suggest_ function on Optuna trials
-            params = {
-                args[0] if len(args) > 0 else kwargs["name"]: getattr(
-                    ot_trial, fn)(*args, **kwargs)
-                for (fn, args, kwargs) in self._space
-            }
+        # getattr will fetch the trial.suggest_ function on Optuna trials
+        params = {
+            args[0] if len(args) > 0 else kwargs["name"]: getattr(
+                ot_trial, fn)(*args, **kwargs)
+            for (fn, args, kwargs) in self._space
+        }
```

Comment:
Unfortunately this has to stay - `trial.params` is apparently only initialized once the methods are called

Reply:
You might be able to make use of the `fixed_distributions` parameter of `Study.ask` ([docs](https://optuna.readthedocs.io/en/latest/reference/generated/optuna.study.Study.html#optuna.study.Study.ask)) which will be introduced in Optuna's next v2.6. It should be coming out in a week or so. I'm not familiar with the implementation details here so please correct me if I'm wrong. I'd imagine you could reimplement `convert_search_space` to construct this dictionary argument.

Thanks for staying updating with our ask and tell API by the way.

Reply:
The `fixed_distributions` parameter sounds exactly like what we want - we'll probably integrate it shortly after the next release!